src/document.rs

```diff
@@ -53,6 +57,7 @@ impl Document {
     pub fn new() -> Self {
         Self {
             version: "1.4".to_string(),
+            binary_mark: vec![0, 0, 0, 0],
```

Comment:
I just noticed that you set the default binary mark to something invalid, which makes no sense.
I would propose to set this to `"��".as_bytes()`. This is two bytes longer than it needs to be, but has the advantage of being human-readable.
(you could also write `b"\xEF\xBF\xBD\xEF\xBF\xBD" if you find it more visually pleasing).

Alternatively, we could use two shorter UTF-8 characters, such as `"αα".as_bytes()`. This would save us two bytes while still being human-readable. 
I would prefer the replacement character, because its intention is more clear than some other random UTF-8 characters.